fission-web-server/library/Fission/Web/Server/Internal/Tasks.hs

```diff
@@ -45,7 +54,71 @@ deleteByUsername userNameTxt =
         Nothing                -> error "User doesn't exist"
 
 ---
+syncDNS :: Server()
+syncDNS = do
+  users <- getUserData
+  apps <- getAppData
+  mapM_ setUserDNS users
+  mapM_ setAppDNS apps
```

Comment:
This totally works! No change required _at all_, but I find that `forM_` reads nicely in these situations:

```haskell
-- Actually any traversable, not just lists
mapM :: Monad m => (a -> m ()) -> [m a] -> m ()
forM :: Monad m => [m a] -> (a -> m ()) -> m ()

syncDNS :: Server ()
syncDNS = do
  users <- getUserData
  apps  <- getAppData
  forM_ users userUserDNS
  forM_ apps  setAppDNS
```

In English: "for the apps, set app DNS"

Just for completeness, there are other phrasings...

```haskell
syncDNS :: Server ()
syncDNS = do
  forM_ users =<< getUserData
  forM_ apps  =<< getAppData
```

...but I think that your existing version where you run the queries first is clearer